bats/tests/helpers/kubernetes.bash

```diff
@@ -47,6 +47,28 @@ wait_for_kube_deployment_available() {
     try assert_kube_deployment_available "$@"
 }
 
+traefik_ip() {
+    local jsonpath='jsonpath={.status.loadBalancer.ingress[0].ip}'
+    run --separate-stderr kubectl get service traefik --namespace kube-system --output "$jsonpath"
+    assert_success || return
+    assert_output || return
+    echo "$output"
+}
+
+traefik_hostname() {
+    if is_windows; then
+        local ip
+        ip=$(traefik_ip) || return
+        echo "${ip}.sslip.io"
+    else
+        echo "localhost"
+    fi
+}
+
+wait_for_traefik() {
+    try traefik_ip
```

Comment:
We still use traefik, we just don't use the IP address of the service because it is the internal Slirp address and not routable from the host (because we are running non-admin installs).

So we don't need traefik to be ready to call `traefik_hostname`, but we will still need it to be ready before the `curl` can succeed.

So to keep the tests simpler I would unconditionally wait for traefik to be ready. If you are concerned about saving some time and running stuff in parallel, then we could move the `wait_for_traefik` step just before we call `traefik_hostname` the first time, but I don't think it makes much difference.

I also expect to remove much of the traefik use and replace it with direct port forwarding in the future.

Reply:
Either way, the `wait_for_traefik` function should wait unconditionally, so it always does what the function name implies. Any conditions need to be applied on the call site.